airflow/www/static/js/cluster-activity/nav/FilterBar.tsx

```diff
@@ -0,0 +1,101 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* global moment */
+
+import { Box, Button, Flex, Input, Text } from "@chakra-ui/react";
+import React from "react";
+
+import { useTimezone } from "src/context/timezone";
+import {
+  isoFormatWithoutTZ,
+  formatDuration,
+  getDuration,
+} from "src/datetime_utils";
+import useFilters from "src/cluster-activity/useFilters";
+
+const FilterBar = () => {
+  const { filters, onStartDateChange, onEndDateChange, clearFilters } =
+    useFilters();
+
+  const { timezone } = useTimezone();
+  const startDate = moment(filters.startDate);
+  const endDate = moment(filters.endDate);
+  const formattedStartDate = startDate.tz(timezone).format(isoFormatWithoutTZ);
+  const formattedEndDate = endDate.tz(timezone).format(isoFormatWithoutTZ);
+
+  const inputStyles = { backgroundColor: "white", size: "lg" };
+
+  return (
+    <Flex
+      backgroundColor="blackAlpha.200"
+      mb={4}
+      px={4}
+      py={5}
+      justifyContent="space-between"
+      flexWrap="wrap"
+      gap={4}
+    >
+      <Flex justifyContent="space-between" flexWrap="wrap" gap={4}>
+        <Box px={2}>
+          <Text fontSize="sm" as="b" position="absolute" mt="-14px" ml={1}>
+            Start Date
+          </Text>
+          <Input
+            {...inputStyles}
+            type="datetime-local"
+            value={formattedStartDate || ""}
+            onChange={(e) => onStartDateChange(e.target.value)}
+          />
+        </Box>
+        <Box px={2}>
+          <Text fontSize="sm" as="b" position="absolute" mt="-14px" ml={1}>
+            End Date
+          </Text>
+          <Input
+            {...inputStyles}
+            type="datetime-local"
+            value={formattedEndDate || ""}
+            onChange={(e) => onEndDateChange(e.target.value)}
+          />
+        </Box>
+        <Flex alignItems="center">
+          <Text whiteSpace="nowrap">
+            over the last{" "}
+            {formatDuration(getDuration(formattedStartDate, formattedEndDate))}
+          </Text>
+        </Flex>
+      </Flex>
+      <Flex px={2}>
+        <Button
+          colorScheme="cyan"
+          aria-label="Reset filters"
+          background="white"
+          variant="outline"
+          onClick={clearFilters}
+          size="lg"
+        >
+          Clear Filters
```

Comment:
Should we only show this when the start/end are not the default?

Reply:
I tried to keep this consistent with the filter bar in the Grid.
![image](https://github.com/apache/airflow/assets/14861206/72222358-3d34-449a-8aaf-743c81cee95e)

Also having elements pop in and out of the page (especially buttons) feels a little weird IMHO. We could grey it out but we would need to do the same in the grid.